### PR TITLE
log on start if its a gobal dep or a local dep which is missing

### LINKED ISSUE #1577: Differentiate error message for dependency checks for local/globalDependencies
... to make more clear what the reason is.

For "local deps" we could just add a "on this host" or such to the error message

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #1577

### CODE REVIEW

File: packages/controller/main.js [hunk 1]
```diff
@@ -3014,73 +3021,68 @@ function checkVersion(id, name, version, instances) {
...
-    });
+    // check local dependencies: required adapter must be installed on the same host
+    try {
+        for (const dep of Object.keys(deps)) {
+            checkVersion(dep, deps[dep], instances);
+        }
+    } catch (e) {
+        logger.error(`${hostLogPrefix} startInstance ${id} [sameHostDependency]: ${e.message}`);
+        logger.debug(`${hostLogPrefix} startInstance ${id} [sameHostDependency]: ${JSON.stringify(deps)}`);
+        throw new Error();
```
Comment (Apollon77) [inline packages/controller/main.js:3073]: Empty error? why not simply rethrow original one?

File: packages/controller/main.js [hunk 2]
```diff
@@ -3014,73 +3021,68 @@ function checkVersion(id, name, version, instances) {
...
+
+    // check global dependencies: required adapter must be NOT installed on the same host
+    try {
+        for (const gDep of Object.keys(globalDeps)) {
+            checkVersion(gDep, globalDeps[gDep], globInstances);
+        }
+    } catch (e) {
+        logger.error(`${hostLogPrefix} startInstance ${id} [globalDependency]: ${e.message}`);
+        logger.debug(`${hostLogPrefix} startInstance ${id} [globalDependency]: ${JSON.stringify(globalDeps)}`);
+        throw new Error();
```
Comment (Apollon77) [inline packages/controller/main.js:3084]: same here

File: packages/controller/main.js (hunk 1 above)
Comment (foxriver76) [reply packages/controller/main.js:3073]: Was like this before but we can change

File: packages/controller/main.js (hunk 2 above)
Comment (foxriver76) [reply packages/controller/main.js:3084]: Same response

**Pull request merged**